generators/pdf_to_text/pdf_to_text/code_snippet_common.md

```diff
@@ -0,0 +1,23 @@
+```python 
+from PyPDF2 import PdfReader
+
+def pdf_to_text(pdf):
+    reader = PdfReader(pdf)
+    number_of_pages = len(reader.pages)
+    number = 0
+    output = ""
+    while number < number_of_pages:
+        output += "".join(reader.pages[number].extract_text())
+        number +=1
+    return output
```

Comment:
Would it make more sense to retrieve the number of pages and then just iterate through all of them? 

```suggestion
def pdf_to_text(pdf):
    reader = PdfReader(pdf)
    return "".join("".join(page.extract_text()) for page in reader.pages)
```